Babylon/commands/azure/adt/instance/create.py

```diff
@@ -53,56 +55,73 @@ def create(
+        logger.error(f"Failed to create ADT instance '{name}': {error_message[0]}")
         return CommandResponse.fail()
 
     # Long-running operations return a poller object; calling poller.result()
     # waits for completion.
     adt_creation_result = poller.result()
-    logger.info(f"Successfully created digital twins instance {adt_creation_result.name}")
-
-    logger.info(f"Adding role assignment to the created instance {adt_creation_result.name}...")
+    adt_host_name = f"https://{adt_creation_result.host_name}"
 
+    logger.info(f"Adding role assignment to the created instance {adt_host_name}...")
     #  Integrated Azure role
-    adt_data_owner_role_id = "bcd981a7-7f74-457b-83e1-cceb9e632ffe"
-
-    scope = ("/subscriptions/" + azure_subscription + "/resourceGroups/" + resource_group_name +
-             "/providers/Microsoft.DigitalTwins/digitalTwinsInstances/" + adt_instance_name)
-
+    adt_data_owner_role_id = env.azure['roles']['adt_owner']
+    scope = ("/subscriptions/" + azure_subscription + "/resourceGroups/" + context['azure_resource_group_name'] +
+             "/providers/Microsoft.DigitalTwins/digitalTwinsInstances/" + name)
     authorization_client = AuthorizationManagementClient(
         credential=get_azure_credentials(),
         subscription_id=azure_subscription,
     )
 
     try:
-        role_assignment = authorization_client.role_assignments.create(
+        authorization_client.role_assignments.create(
             scope,
             str(uuid.uuid4()),
             {
                 "roleDefinitionId":
                     "/subscriptions/" + azure_subscription + "/providers/Microsoft.Authorization/roleDefinitions/" +
                     adt_data_owner_role_id,
                 "principalId":
-                    principal_id,
+                    context['babylon_principal_id'],
                 "principalType":
                     "ServicePrincipal",
             },
         )
     except Exception as e:
-        logger.error(f"Failed to assign a new role to ADT instance '{adt_instance_name}': {e}")
+        logger.error(f"Failed to assign a new role to ADT instance '{name}': {e}")
 
-    return CommandResponse.success({
-        "adt_instance": adt_creation_result.name,
-        "role_assignment": role_assignment,
-    })
+    try:
+        authorization_client.role_assignments.create(
+            scope,
+            str(uuid.uuid4()),
+            {
+                "roleDefinitionId":
+                    "/subscriptions/" + azure_subscription + "/providers/Microsoft.Authorization/roleDefinitions/" +
+                    adt_data_owner_role_id,
+                "principalId":
+                    context['platform_principal_id'],
+                "principalType":
+                    "ServicePrincipal",
+            },
+        )
+    except Exception as e:
+        logger.error(f"Failed to assign a new role to ADT instance '{name}': {e}")
```

Comment:
Maybe we could save some lines of code like this:
```python
    for principal_id in ['babylon_principal_id', 'platform_principal_id']:
        try:
            authorization_client.role_assignments.create(
                scope,
                str(uuid.uuid4()),
                {
                    "roleDefinitionId":
                        "/subscriptions/" + azure_subscription + "/providers/Microsoft.Authorization/roleDefinitions/" +
                        adt_data_owner_role_id,
                    "principalId":
                        context[principal_id],
                    "principalType":
                        "ServicePrincipal",
                },
            )
        except Exception as e:
            logger.error(f"Failed to assign a new role to ADT instance '{name}': {e}")
```

Reply:
yes, thanks. done!